### PR TITLE
feat: add group/type to resource slog

### PR DESCRIPTION
**Pull request by zhongwencool**
Fixes https://emqx.atlassian.net/browse/EMQX-12633
1. Add resource_type for every resource, add to log tag meta.
2. Log as info(not warning) if is dry_run failed.

Release version: v/e5.7.2

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

**Comment by zmstone** [comment]
`resource_type()` is a module, we can add a `name/0` callback to the `emqx_resource` behavior, this should avoid changing the APIs where ResourceType is already passed as an argument.

**Comment by zhongwencool** [comment]
> `resource_type()` is a module, we can add a `name/0` callback to the `emqx_resource` behavior, this should avoid changing the APIs where `ResourceType` is already passed as an argument.

done.

File: apps/emqx_bridge_hstreamdb/src/emqx_bridge_hstreamdb_connector.erl [hunk 1]
```diff
@@ -44,6 +45,8 @@
 
 %% -------------------------------------------------------------------------------------------------
 %% resource callback
+resource_type() -> hstreamsdb.
```
Comment (zmstone) [inline apps/emqx_bridge_hstreamdb/src/emqx_bridge_hstreamdb_connector.erl:48]:
```suggestion
resource_type() -> hstream.
```

File: apps/emqx_postgresql/src/emqx_postgresql.erl [hunk 2]
```diff
@@ -120,6 +121,8 @@ adjust_fields(Fields) ->
     ).
 
 %% ===================================================================
+resource_type() -> postgresql.
```
Comment (zmstone) [inline apps/emqx_postgresql/src/emqx_postgresql.erl:124]:
```suggestion
resource_type() -> postgres.
```

File: apps/emqx_bridge_hstreamdb/src/emqx_bridge_hstreamdb_connector.erl (hunk 1 above)
Comment (zhongwencool) [reply apps/emqx_bridge_hstreamdb/src/emqx_bridge_hstreamdb_connector.erl:48]: https://github.com/emqx/emqx/blob/master/apps/emqx_bridge/src/schema/emqx_bridge_enterprise.erl#L105
We utilize `hstreamdb` in the API; keep them consistent.

File: apps/emqx_postgresql/src/emqx_postgresql.erl (hunk 2 above)
Comment (zhongwencool) [reply apps/emqx_postgresql/src/emqx_postgresql.erl:124]: https://github.com/emqx/emqx/blob/master/apps/emqx_bridge/src/schema/emqx_bridge_enterprise.erl#L117
We utilize `pgsql` in the API; keep them consistent.

**Review by zmstone — APPROVED** [review]

**Comment by emqxqa** [comment]
[TestExecution](https://emqx.atlassian.net/browse/QA-3226)

**Pull request merged**